### PR TITLE
Refactoring external link directory

### PR DESCRIPTION
**Pull request by yerramshilpa**
## Type of Change (Select One and Apply Github Label)

- [ ] Bug fix (non-breaking change which fixes an issue) -> Apply bugfix label
- [x] New feature (non-breaking change which adds functionality) -> Apply enhancement label
- [ ] Breaking change (fix or feature that would cause existing functionality to change) -> Apply breaking label

## Which browsers have you tested?
- [x] Internet Explorer 11
- [ ] Edge
- [x] Chrome
- [ ] Firefox
- [x] Safari

## Checklist:

- [ ] I have read the (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md] document.
- [ ] My code passes the automated linter.
- [ ] This code has been reviewed by another team member and passes the reviewer checklist found in (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md]
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My code is 508 compliant as tested by AMP and JAWS
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/packages/components/src/lib/external-link/external-link.directive.ts [hunk 1]
```diff
@@ -1,52 +1,56 @@
...
-  @HostListener('click', ['$event'])
-  click(event: Event) {
-    window.location.href = this.href;
-  }
-
   public ngOnChanges() {
+    this.hrefAttr = this.href;
+    this.relAttr = 'noopener';
+    this.targetAttr = '_blank';
     if (!this.isExternalLink) {
```
Comment (Bibek-Regmi) [inline libs/packages/components/src/lib/external-link/external-link.directive.ts:35]: This condition checks for whether something is an external link or not - Those attributes should only be set if the link is external right? Should they then be in the else statement?

Comment (yerramshilpa) [reply libs/packages/components/src/lib/external-link/external-link.directive.ts:35]: It should update if it is external only, moved code

File: libs/packages/components/src/lib/external-link/external-link.directive.ts [hunk 2]
```diff
@@ -1,52 +1,56 @@
...
       return;
-    }
-    if (!this.hideIcon) {
-      this.createIcon();
+    } else {
+      if (!this.hideIcon) {
+        this.createIcon();
+        this.hrefAttr = this.href;
+        this.relAttr = 'noopener';
+        this.targetAttr = '_blank';
```
Comment (Bibek-Regmi) [inline libs/packages/components/src/lib/external-link/external-link.directive.ts:39]: Per our discussion - we do still want these attributes even when hideIcon is true.

Comment (yerramshilpa) [reply libs/packages/components/src/lib/external-link/external-link.directive.ts:39]: Yes

**Review by Bibek-Regmi — APPROVED** [review]

**Pull request merged**